### ISSUE TITLE
format-NAME-prefix-fg overrides format-NAME-fg

### ISSUE THREAD

**Issue by alexciarlillo** [jaagr/polybar#544]
Setting foreground/background colors on the prefix seem to override the "regular" set values. e.g.:
```

format-volume-prefix-background = ${colors.white}
format-volume-prefix-foreground = ${colors.black}
format-volume-background = ${colors.black} // these 
format-volume-foreground = ${colors.white} // dont get applied
```

It seems that the colors are not being reset to the proper values after the prefix is printed. Or am I just doing something wrong?

**Comment by jaagr** [comment]
Could be a bug. Could you post one output line from running polybar with the `-s` flag?

**Comment by alexciarlillo** [comment]
I created a simplified bar to test this. Here is the config and the -s output:

config:
```
[colors]
black = ${xrdb:color0:#282828}
white = ${xrdb:color15:#ebdbb2}

[bar/test]
width = 100%
height = 25
radius = 0
fixed-center = true

background = ${colors.background}
foreground = ${colors.foreground}

font-0 = "Hack:size=10;2"
font-1 = "FontAwesome:size=10;2"
font-2 = "Hack:size=20;3"

modules-left =
modules-center = voltest
modules-right =

[module/voltest]
type = internal/volume

format-volume-padding = 0
format-volume-background = ${colors.white}
format-volume-foreground = ${colors.black}
format-volume-prefix-background = ${colors.black}
format-volume-prefix-foreground = ${colors.white}
format-volume-prefix = " prefix "
format-volume = "<label-volume> <bar-volume> "
label-volume = " "

format-muted = <label-muted>
label-muted = " muted "

bar-volume-width = 8
bar-volume-gradient = false
bar-volume-indicator = 
bar-volume-indicator-font = 2
bar-volume-fill = ─
bar-volume-fill-font = 2
bar-volume-empty = ─
bar-volume-empty-font = 2
bar-volume-empty-foreground = ${colors.white}
```

output:
```
ciarlill@mervyn2 ~ $ polybar test -s
warn: No monitor specified, using "eDP1"
%{c}%{A1:volmute: A4:volup: A5:voldown: B#ebdbb2 F#282828 B#282828 F#ebdbb2} prefix %{B- F-}  %{T2}──────%{T2}%{T- F#ebdbb2 T2}─%{T- F-} %{F- B- A A A}
```

**Comment by jaagr** [comment]
Thanks I'll have a closer look at it after work.

**Comment by jaagr** [comment]
We'll have to create a value stack for some formatting tags to be able to pop to the previous value because now the resetting (e.g. `%{F-}`) only sets the default value defined for the bar.

**Comment by alexciarlillo** [comment]
That makes sense. Thanks for looking into it. I have found workarounds for what I was trying to accomplish using inline manual formatting. A little verbose, but the approach should work for anyone else that runs into this.

**Comment by jaagr** [comment]
@alexciarlillo I implemented a color value stack. Please try it out.

**Comment by patrick96** [comment]
I think this should be fixed as well with #729. @alexciarlillo could you please test, if this is now fixed for you, it seems to be fixed on my end

**Comment by alexciarlillo** [comment]
Yes. I was able to do a bit of testing last night and it appears to work fine. Thanks for the great work!
